kexoqs.c

```diff
@@ -127,15 +136,24 @@ static int kex_kem_generic_dec(OQS_KEM *kem,
     r = SSH_ERR_ALLOC_FAIL;
     goto out;
   }
-  if ((r = sshbuf_reserve(buf, kem->length_shared_secret, &kem_key)) != 0)
+  if ((kem_key = malloc(kem->length_shared_secret)) == NULL) {
+    r = SSH_ERR_ALLOC_FAIL;
     goto out;
+  }
   if (OQS_KEM_decaps(kem, kem_key, ciphertext, kex->oqs_client_key) != OQS_SUCCESS) {
+    r = SSH_ERR_LIBCRYPTO_ERROR;
     goto out;
   }
+  if ((r = sshbuf_put_string(buf, kem_key, kem->length_shared_secret)) != 0)
+    goto out;
```

Comment:
```suggestion
  if ((r = sshbuf_put_string(buf, kem_key, kem->length_shared_secret)) != 0) {
    goto out;
  }
```

Reply:
I'm fine either way. I went without the braces here because the one-line body if at line 93 also omits them, and I was trying to be consistent with existing style.

Reply:
I don't like that style because it leads to bugs, e.g. Apple's [goto fail bug](https://www.imperialviolet.org/2014/02/22/applebug.html).  I wouldn't go so far as to change all occurrences within the OpenSSH code, but in code we add, I'd prefer to include it.

Reply:
Fine by me. I prefer that style too, but as I said, the existing code had it the other way. I'll add another commit with this change.